File: yohan-backend/app/schemas/weather.py
from pydantic import BaseModel
from typing import List
from datetime import datetime, date

class CurrentWeather(BaseModel):
    temp: float
    feels_like: float
    humidity: int
    wind_speed: float
    description: str
    sunrise: datetime
    sunset: datetime

class ForecastDay(BaseModel):
    date: date
    max_temp: float
    min_temp: float
    description: str

class HourlyForecast(BaseModel):
    time: datetime
    temp: float
    description: str

class WeatherData(BaseModel):
    current: CurrentWeather
    hourly: List[HourlyForecast]
    daily: List[ForecastDay]

File: yohan-frontend/src/types/weather.ts
// TypeScript interfaces that mirror the backend Pydantic weather schemas

export interface CurrentWeather {
  temp: number;
  feels_like: number;
  humidity: number;
  wind_speed: number;
  description: string;
  sunrise: string; // ISO datetime string
  sunset: string; // ISO datetime string
}

export interface ForecastDay {
  date: string; // ISO date string
  max_temp: number;
  min_temp: number;
  description: string;
}

export interface HourlyForecast {
  time: string; // ISO datetime string
  temp: number;
  description: string;
}

export interface WeatherData {
  current: CurrentWeather;
  hourly: HourlyForecast[];
  daily: ForecastDay[];
  location?: string; // Optional location field
}
